### PR TITLE
[MAINTENANCE] Improve typing in OSS GE (using Optional type)

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Changes proposed in this pull request:
- This PR is part of incremental work we intend to do around [Typing](https://docs.python.org/3/library/typing.html) in OSS GE. 
- In this PR, we want to ensure that `bool` type is not kept as Optional, since we usually don't want them to be `None` along with `True`/`False`.
- In this PR, we intend to change the following parameter with `bool` from this

`fetch_all: Optional[bool] = False,`

to this

`fetch_all: bool = False`

- IPV-251
- Thank you @cdkini for the amazing `grep` script that was used to generate this PR.  Which I've copied here for future reference. 

```
grep "Optional\[bool\] = True" -rl great_expectations/ | xargs sed -i '' 's/Optional\[bool\] = True/bool = True/g'
grep "Optional\[bool\] = False" -rl great_expectations/ | xargs sed -i '' 's/Optional\[bool\] = False/bool = False/g'
```

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

**Review by alexsherstinsky — APPROVED** [review]
LGTM!

**Comment by alexsherstinsky** [comment]
@Shinnnyshinshin We seem to have one of the upgrade utilities failing a test.  I will rerun -- just in case it is an intermittent issue, but if it fails again, we might need to troubleshoot tomorrow...

File: great_expectations/cli/upgrade_helpers/upgrade_helper_v13.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def __init__(
         self,
         data_context: Optional[DataContext] = None,
         context_root_dir: Optional[str] = None,
-        update_version: Optional[bool] = False,
+        update_version: bool = False,
```
Comment (alexsherstinsky) [inline great_expectations/cli/upgrade_helpers/upgrade_helper_v13.py:24]: @Shinnnyshinshin This seems like a proper change, yet a couple of tests started failing...  Not sure why.

**Comment by alexsherstinsky** [comment]
@Shinnnyshinshin Tests failed the same way after a rerun -- I suspect that if they pass locally, it is something to do with the slight differences in the operating systems between our local development environment and the virtual machines in Azure.  Thanks!

**Comment by alexsherstinsky** [comment]
@Shinnnyshinshin I fixed the fixtures and committed them to your branch in GitHub -- hope you do not mind.  If all tests pass, please feel free to merge at will. Thanks!

**Comment by cdkini** [comment]
LGTM thanks for this 🎉

**Review by cdkini — APPROVED** [review]

**Pull request merged**